x/airdrop/keeper/keeper.go

```diff
@@ -0,0 +1,140 @@
+package keeper
+
+import (
+	"fmt"
+
+	errorsmod "cosmossdk.io/errors"
+	"cosmossdk.io/log"
+	"github.com/cosmos/cosmos-sdk/codec"
+	"github.com/cosmos/cosmos-sdk/store/prefix"
+	storetypes "github.com/cosmos/cosmos-sdk/store/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+
+	"github.com/Stride-Labs/stride/v22/x/airdrop/types"
+)
+
+type (
+	Keeper struct {
+		cdc      codec.BinaryCodec
+		storeKey storetypes.StoreKey
+		logger   log.Logger
+
+		accountKeeper types.AccountKeeper
+		bankKeeper    types.BankKeeper
+	}
+)
+
+func NewKeeper(
+	cdc codec.BinaryCodec,
+	storeKey storetypes.StoreKey,
+	logger log.Logger,
+
+	accountKeeper types.AccountKeeper,
+	bankKeeper types.BankKeeper,
+) Keeper {
+	return Keeper{
+		cdc:      cdc,
+		storeKey: storeKey,
+		logger:   logger,
+
+		accountKeeper: accountKeeper,
+		bankKeeper:    bankKeeper,
+	}
+}
+
+// Logger returns a module-specific logger.
+func (k Keeper) Logger() log.Logger {
+	return k.logger.With("module", fmt.Sprintf("x/%s", types.ModuleName))
+}
+
+func (k Keeper) Claim(ctx sdk.Context, claimer string) error {
+	claimerAccount, err := sdk.AccAddressFromBech32(claimer)
+	if err != nil {
+		return errorsmod.Wrapf(sdkerrors.ErrInvalidAddress, "invalid address (%s)", claimer)
+	}
+
+	// TODO implement logic
+
+	return nil
+}
+
+func (k Keeper) ClaimAndStake(ctx sdk.Context, claimer string) error {
+	claimerAccount, err := sdk.AccAddressFromBech32(claimer)
+	if err != nil {
+		return errorsmod.Wrapf(sdkerrors.ErrInvalidAddress, "invalid address (%s)", claimer)
+	}
+
+	// TODO implement logic
+
+	return nil
+}
+
+func (k Keeper) ClaimEarly(ctx sdk.Context, claimer string) error {
+	claimerAccount, err := sdk.AccAddressFromBech32(claimer)
+	if err != nil {
+		return errorsmod.Wrapf(sdkerrors.ErrInvalidAddress, "invalid address (%s)", claimer)
+	}
+
+	// TODO implement logic
+
+	return nil
+}
+
+func (k Keeper) GetAirdropRecords(ctx sdk.Context) []types.AirdropRecord {
+	// TODO add pagination?
+	store := prefix.NewStore(ctx.KVStore(k.storeKey), types.AirdropRecordsKeyPrefix)
```

Comment:
nbd at all, but for future reference it's kinda nice to push these to a separate PR so comments are a little more manageable on each!

Reply:
i realize now that this was probably auto generated from ignite haha

Reply:
no this was all me 😅